core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -322,25 +327,26 @@ public void requestSpace(long sessionId, long blockId, long additionalBytes)
     }
     // NOTE: a temp block is only visible to its own writer, unnecessary to acquire
     // block lock here since no sharing
-    try (LockResource r = new LockResource(mMetadataWriteLock)) {
-      TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    TempBlockMeta tempBlockMeta;
+    try (LockResource r = new LockResource(mMetadataReadLock)) {
+      tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    }
 
-      StorageDirView allocationDir = allocateSpace(sessionId,
-          AllocateOptions.forRequestSpace(additionalBytes, tempBlockMeta.getBlockLocation()));
-      if (!allocationDir.toBlockStoreLocation().equals(tempBlockMeta.getBlockLocation())) {
-        // If reached here, allocateSpace() failed to enforce 'forceLocation' flag.
-        throw new IllegalStateException(
-            String.format("Allocation error: location enforcement failed for location: %s",
-                allocationDir.toBlockStoreLocation()));
-      }
+    StorageDirView allocationDir = allocateSpace(sessionId,
+        AllocateOptions.forRequestSpace(additionalBytes, tempBlockMeta.getBlockLocation()));
+    if (!allocationDir.toBlockStoreLocation().equals(tempBlockMeta.getBlockLocation())) {
```

Comment:
what if the tempBlockMeta got updated by other threads?  looks like you're shifting these lines out from the resource lock.  any particular reason for you doing this?